versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseMergeTransplantIndividual.java

```diff
@@ -87,16 +87,18 @@ MergeResult<Commit> individualCommits(
 
       verifyMergeTransplantCommitPolicies(targetParentIndex, sourceCommit);
 
+      List<Obj> objsToStore = new ArrayList<>();
       CommitObj newCommit =
-          createMergeTransplantCommit(mergeBehaviorForKey, keyDetailsMap, createCommit);
+          createMergeTransplantCommit(
+              mergeBehaviors, keyDetailsMap, createCommit, objsToStore::add);
 
       if (!indexesLogic.commitOperations(newCommit).iterator().hasNext()) {
         // No operations in this commit, skip it.
         continue;
       }
 
       CommitLogic commitLogic = commitLogic(persist);
-      boolean committed = commitLogic.storeCommit(newCommit, emptyList());
+      boolean committed = commitLogic.storeCommit(newCommit, objsToStore);
```

Comment:
We should probably not call this method if/when `keyDetailsMap` contains conflicts.